File: android/Reguerta/app/src/main/java/com/reguerta/user/domain/profiles/SharedProfile.kt
package com.reguerta.user.domain.profiles

data class SharedProfile(
    val userId: String,
    val familyNames: String,
    val photoUrl: String?,
    val about: String,
    val updatedAtMillis: Long,
) {
    val hasVisibleContent: Boolean
        get() = familyNames.isNotBlank() || !photoUrl.isNullOrBlank() || about.isNotBlank()
}

File: ios/Reguerta/Reguerta/Domain/Profiles/SharedProfile.swift
import Foundation

struct SharedProfile: Identifiable, Equatable, Sendable {
    var id: String { userId }
    let userId: String
    let familyNames: String
    let photoUrl: String?
    let about: String
    let updatedAtMillis: Int64

    var hasVisibleContent: Bool {
        !familyNames.isEmpty || !(photoUrl?.isEmpty ?? true) || !about.isEmpty
    }
}
